src/components/AssignBusToDriver/assignDriver.js

```diff
@@ -0,0 +1,189 @@
+import { SkeletonTable } from '../../skeletons/skeletonTable';
+import { useDispatch, useSelector } from 'react-redux';
+import '../AssignBusToDriver/assign.scss';
+import { useTranslation } from 'react-i18next';
+import {getDriversAction, getBusAction, assignDrivertoBusAction, unAssignDriverToBusAction} from '../../redux/actionCreators/assignAction';
+
+
+
+
+
+const AssignDriver = () => {
+  const { t } = useTranslation();
+  const [showModal, setShowModal] = useState(false);
+  const [dropDownAssign, setDropDownAssign] = useState('');
+  const [clickedDriverID, setClickedDriverId] = useState(null);
+
+
+  const dispatch = useDispatch();
+
+  const state = useSelector((state) => state.getDriverReducer.data);
+  const getBusState = useSelector((state) => state.getBusReducer);
+
+  const assignDriver = () => {
+    dispatch(assignDrivertoBusAction(dropDownAssign,clickedDriverID));
+    setShowModal(false);
+   
+  };
+
+  const unassignDriver = () => {
+    dispatch(unAssignDriverToBusAction(clickedDriverID));
+    setShowModal(false);
+    
+  };
+
+
+
+  useEffect(()=>{
+    dispatch(getDriversAction());
+    dispatch(getBusAction());
+  }, []);
+
+
+  
+return(
+
+    <div className="tbl">
+        
+
+    <table className="text-left w-full shadow overflow-hidden rounded border-b border-gray-200 sm:text-left" data-testid="assignTable">
+		<thead className="bg-green-200 flex text-black w-full">
+			<tr className="flex w-full mb-1">
+				<th className="p-4 w-1/4">{t('assignDriver.firstName')}</th>
+				<th className="p-4 w-1/4">{t('assignDriver.lastName')}</th>
+				<th className="p-4 w-1/4">{t('assignDriver.PlateNo')}</th>
+        <th className="p-4 w-1/4">{t('assignDriver.Status')}</th>
+			</tr>
+		</thead>
+		<tbody className="bg-grey-light flex flex-col items-center  overflow-y-scroll w-full">
+      
+    {!state &&
+                [1, 2, 3, 4].map((skeleton) => (
+                  <tr key={skeleton} className="flex w-full mb-3 hover:bg-orange-100">
+                    <td className='p-4 w-1/4'>
+                      <SkeletonTable />
+                    </td>
+                    <td className='p-4 w-1/4'>
+                      <SkeletonTable />
+                    </td>
+                    <td className='p-4 w-1/4'>
+                      <SkeletonTable />
+                    </td>
+                    <td className='p-4 w-1/4'>
+                      <SkeletonTable />
+                    </td>
+                    
+                  </tr>
+                ))}
+
+      { state && state.map((data) => (
+                  <tr key={data.id} className="flex w-full mb-3 hover:bg-orange-100"
+                  style={{ transition: "all .15s ease" }}
+                  onClick={() => {setShowModal(true);
+                                  setClickedDriverId(data.id)
+                          }}
+                  data-testid="assignData">
+
+                    <td className='p-4 w-1/4'>
+                      {data.firstName}
```

Comment:
let' use es6 destructuring to get firstName, lastName and the busId . i.e `const {firstName, LastName, budId} = data`